vars/folioSchemaCompareUtils.groovy

```diff
@@ -0,0 +1,291 @@
+                password : kubectl.getSecretValue(rancher_project_name, 'db-connect-modules', 'DB_PASSWORD'),
+                host     : kubectl.getSecretValue(rancher_project_name, 'db-connect-modules', 'DB_HOST'),
+                user     : kubectl.getSecretValue(rancher_project_name, 'db-connect-modules', 'DB_USERNAME'),
+                db       : kubectl.getSecretValue(rancher_project_name, 'db-connect-modules', 'DB_DATABASE'),
+                port     : kubectl.getSecretValue(rancher_project_name, 'db-connect-modules', 'DB_PORT')
+            ]
+
+            def teamAssignment = getTeamAssignment()
+
+            def atlasPodName = "atlas"
+            kubectl.runPodWithCommand(rancher_project_name, atlasPodName, 'arigaio/atlas:0.10.1-alpine')
+
+            def psqlPodName = "psql-client"
+            kubectl.runPodWithCommand(rancher_project_name, psqlPodName, 'andreswebs/postgresql-client')
+
+            def srcSchemasList = getSchemaTenantList(rancher_project_name, psqlPodName, tenant_id, psqlConnection)
+            def dstSchemasList = getSchemaTenantList(rancher_project_name, psqlPodName, tenant_id_clean, psqlConnection)
+
+            def groupedValues = [:]
+            def uniqueValues = []
+
+            srcSchemasList.each { srcValue ->
+                def currentSuffix = srcValue.split('_')[1..-1].join('_')
+                dstSchemasList.each { dstValue ->
+                    def newSuffix = dstValue.split('_')[1..-1].join('_')
+                    if (currentSuffix == newSuffix) {
+                        groupedValues[srcValue] = dstValue
+                    }
+                }
+                if (!groupedValues.containsKey(srcValue)) {
+                    uniqueValues.add(srcValue)
+                }
+            }
+
+            dstSchemasList.each { dstValue ->
+                def newSuffix = dstValue.split('_')[1..-1].join('_')
+                def alreadyGrouped = false
+                groupedValues.each { srcValue, existingValue ->
+                    def currentSuffix = srcValue.split('_')[1..-1].join('_')
+                    if (newSuffix == currentSuffix) {
+                        alreadyGrouped = true
+                    }
+                }
+                if (!alreadyGrouped) {
+                    uniqueValues.add(dstValue)
+                }
+            }
+
+            if (uniqueValues) {
+                diff.put("Unique schemas", "Please check list of unique Schemas:\n $uniqueValues")
+            }
+
+            groupedValues.each { srcValue, dstValue ->
+                try {
+                    def getDiffCommand = "./atlas schema diff --from 'postgres://${psqlConnection.user}:${psqlConnection.password}@${psqlConnection.host}:${psqlConnection.port}/${psqlConnection.db}?sslmode=disable&search_path=${srcValue}' --to 'postgres://${psqlConnection.user}:${psqlConnection.password}@${psqlConnection.host}:${psqlConnection.port}/${psqlConnection.db}?sslmode=disable&search_path=${dstValue}'"
```

Comment:
In this construction, we have 2 identical strings that differ with only one parameter. 'postgres://${psqlConnection.user}:${psqlConnection.password}@${psqlConnection.host}:${psqlConnection.port}/${psqlConnection.db}?sslmode=disable&search_path'
Maybe better to move it to separate variable and then just concat them?

Reply:
def dbConnectionString = 'postgres://${psqlConnection.user}:${psqlConnection.password}@${psqlConnection.host}:${psqlConnection.port}/${psqlConnection.db}?sslmode=disable&search_path'
                    def getDiffCommand = "./atlas schema diff --from '${dbConnectionString}=${srcValue}' --to '${dbConnectionString}=${dstValue}'"

Reply:
like this style normal ?

Reply:
yes

Reply:
But please be careful) In your example dbConnectionString will not resolve variables, because of single quotes, change to double quotes.

Reply:
works
thanks Dear Oleksandr